src/io/files.js

```diff
@@ -854,11 +861,15 @@ p5.prototype.httpDo = function () {
         type = 'json';
       } else if (request.url.indexOf('xml') !== -1) {
         type = 'xml';
-      } else {
+      } else if (request.url.indexOf('text') !== -1) {
         type = 'text';
+      } else if (request.url.indexOf('array') !== -1) {
+        type = 'array';
+      } else if (request.url.indexOf('integer') !== -1) {
+        type = 'integer';
```

Comment:
This block should not be changed as well for one this is to detect return type from the server and not the datatype the request is sending, for another, detecting if the URL string contains `text`, `array` or `integer` doesn't make sense...